torch_geometric/sampler/neighbor_sampler.py

```diff
@@ -365,57 +373,94 @@ def sample_from_edges(
-                    self.input_type[-1]: seed_dst,
-                }
-                if edge_label_time is not None:
-                    seed_time_dict = {
-                        self.input_type[0]:
-                        scatter_min(edge_label_time, inverse_src)[0],
-                        self.input_type[-1]:
-                        scatter_min(edge_label_time, inverse_dst)[0],
+                if self.disjoint_sampling:
+                    seed_src = edge_label_index[0]
+                    seed_dst = edge_label_index[1]
+                    edge_label_index = torch.arange(
+                        0, num_seed_edges).repeat(2).view(2, -1)
+                    seed_dict = {
+                        self.input_type[0]: seed_src,
+                        self.input_type[-1]: seed_dst,
                     }
+                    if edge_label_time is not None:
+                        seed_time_dict = {
+                            self.input_type[0]: edge_label_time,
+                            self.input_type[-1]: edge_label_time
+                        }
+                else:
+                    seed_src = edge_label_index[0]
+                    seed_src, inverse_src = seed_src.unique(
+                        return_inverse=True)
+                    seed_dst = edge_label_index[1]
+                    seed_dst, inverse_dst = seed_dst.unique(
+                        return_inverse=True)
+                    edge_label_index = torch.stack([inverse_src, inverse_dst],
+                                                   0)
+                    seed_dict = {
+                        self.input_type[0]: seed_src,
+                        self.input_type[-1]: seed_dst,
+                    }
+                    if edge_label_time is not None:
+                        seed_time_dict = {
+                            self.input_type[0]:
+                            scatter_min(edge_label_time, inverse_src)[0],
+                            self.input_type[-1]:
+                            scatter_min(edge_label_time, inverse_dst)[0],
+                        }
 
             else:  # Merge both source and destination node indices:
-                seed_nodes = edge_label_index.view(-1)
-                seed_nodes, inverse = seed_nodes.unique(return_inverse=True)
-                edge_label_index = inverse.view(2, -1)
-                seed_dict = {self.input_type[0]: seed_nodes}
-                if edge_label_time is not None:
-                    tmp = torch.cat([edge_label_time, edge_label_time])
-                    seed_time_dict = {
-                        self.input_type[0]: scatter_min(tmp, inverse)[0]
-                    }
+                if self.disjoint_sampling:
+                    seed_nodes = edge_label_index.view(-1)
+                    edge_label_index = torch.arange(0,
+                                                    2 * num_seed_edges).view(
+                                                        2, -1)
```

Comment:
Don't we need a `repeat` here as well?

Reply:
No because we use `2*num_seed_edges`. The dst nodes have id from `(0,num_seed_edges-1)` and the src nodes of the same type have ids `(num_seed_edges, 2*num_seed_edges-1)`